Mathlib/Computability/Encoding.lean

```diff
@@ -33,9 +33,13 @@ namespace Computability
 
 /-- An encoding of a type in a certain alphabet, together with a decoding. -/
 structure Encoding (α : Type u) where
+  /-- The alphabet of the encoding. -/
   Γ : Type v
+  /-- The encoding function. -/
   encode : α → List Γ
+  /-- The decoding function, which can return `none` for invalid encodings. -/
```

Comment:
*Can* it return none for invalid encodings, or *will* it always do? (If the latter, let's be more precise. I have no idea which of these is the case.)

Reply:
I mean I am no more an expert on this than you, but if you look at the structure, you can see that there is nothing preventing it from returning `some` values for strings that are not valid encodings of anything. 🤷 

Debatably, this should not be the case, and another field should be added to guarantee that invalid encodings decode to return none. It would make it more tedious to prove things, but would make it harder to define a confusing encoder/decoder pair.